src/Authenticator/HttpDigestAuthenticator.php

```diff
@@ -114,20 +112,20 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
      * Gets the digest headers from the request/environment.
      *
      * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
-     * @return array Array of digest information.
+     * @return array|null Array of digest information.
```

Comment:
Why weakinging the return types? Empty array should suffice, should it not?

Reply:
Because null makes sense. Returning an empty array is not much of a digest, is it?
Also parseAuthData() may return null and this method returns null so its typehint was just wrong?

Reply:
All right.